### DISCUSSION TITLE
articles/nginx-configuration

### DISCUSSION THREAD

**Discussion by giscus** [hutusi/hutusi.github.com#159] [Comments]
# articles/nginx-configuration

最近在整理博客站点时发现如果输入错URL不是返回404页面，而是返回500错误，经定位发现是 Nginx 配置的问题。因此周末花了一天时间好好看了下 Nginx 相关配置指导，正好也做一个学习总结。末尾我会说明下这个问题以及修复方法。

https://hutusi.com/articles/nginx-configuration

**Comment by chenyukang** [comment]
啊， 我觉得自己维护服务器很繁琐，
所以现在直接放 Github pages 了，好像也没碰到什么问题.....

**Reply by hutusi** [reply]
github pages 确实方便，唯一的问题就是国内访问速度不太稳定~~

**Comment by ColarXu** [comment]
虽然没看懂（文科生），但看你一直在更新，我也把自己的博客修好了。

**Reply by hutusi** [reply]
独立博主互相勉励，一起加油！💪

**Comment by semyin** [comment]
以前经常在服务器上手动修改nginx配置文件，经常改完报错不通过，后来我就在本地搞了个模板，每次改完传到服务器。因为经常会在服务器跑一些测试的小项目，需要配好多个conf，所以还是觉得不方便，然后就为了偷懒换了第三方图形化配置的工具。

> nginx可视化网页配置工具 https://github.com/cym1102/nginxWebUI

**Reply by hutusi** [reply]
这个工具不错啊，看来有需求就会有解决方案。

**Comment by joker9657** [comment]
typo: 架构 => 价格

**Reply by hutusi** [reply]
谢谢，已fix👍

**Comment by bearfm** [comment]
我放在了Vercel了，速度起飞

**Reply by hutusi** [reply]
Vercel 国内速度有那么快？我有另外个站是放在 Netlify 上了，感觉速度一般~

**Comment by Erven2016** [comment]
我用国内的静态网站托管服务，感觉比云主机省钱（大概每天一分钱，因为流量用量不足按最低 0.01 元计费），hugo 代码放在 GitHub 上， push 后 GitHub Action 会自动构建并推送到托管服务这边，我参考的是这个[配置方法](https://cloud.tencent.com/developer/article/1780127)

主要是省钱又省事，一次配置一劳永逸🤭

**Reply by hutusi** [reply]
谢谢，我来研究下~

**Comment by acegao9527** [comment]
写得特别认真又细致

**Reply by hutusi** [reply]
谢谢😀
